x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -56,19 +59,40 @@ public Match(
             description = "Text you wish to find in the provided field."
         ) Expression matchQuery
     ) {
+        this(source, field, matchQuery, false);
+    }
+
+    private Match(Source source, Expression field, Expression matchQuery, boolean isOperator) {
         super(source, matchQuery, List.of(field, matchQuery));
         this.field = field;
+        this.isOperator = isOperator;
+    }
+
+    public static Match operator(Source source, Expression field, Expression matchQuery) {
+        return new Match(source, field, matchQuery, true);
     }
 
-    private Match(StreamInput in) throws IOException {
-        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class), in.readNamedWriteable(Expression.class));
+    private static Match readFrom(StreamInput in) throws IOException {
+        Source source = Source.readFrom((PlanStreamInput) in);
+        Expression field = in.readNamedWriteable(Expression.class);
+        Expression query = in.readNamedWriteable(Expression.class);
+        boolean isOperator = false;
+        Expression boost = null;
+        Expression fuzziness = null;
+        if (in.getTransportVersion().onOrAfter(TransportVersions.MATCH_OPERATOR_COLON)) {
+            isOperator = in.readBoolean();
```

Comment:
Just a question to confirm that I understand it right - a `TransportVersion` is need here because of isOperator will be needed by the data nodes, is it right? If it is only needed by the coordinator node, perhaps a `TransportVersion` is not needed.

Reply:
I added a `TransportVersion` as we're changing the serialization / deserialization, adding a new class field.

As you're mentioning, the attribute is probably just needed on the coordinator node and thus there's no point in serializing it. But I think we should be consistent with the serialization semantics here unless we have a good reason not to. So I have a couple of questions:

- Is sending the plan to the data nodes the only way of triggering serialization and deserialization via transport in ES|QL? 
- Are there other instances in ES|QL where we're avoiding serialization of attributes based on the knowledge that data nodes are not using that information?

For me, it would be simpler to add the serialization of the attribute just in case (we may add for example a check on the data nodes that needs to error out with this information), but I'm happy to follow standards. 

Let me know and I can remove it, adding a `transient` keyword and a comment so people are aware of this attribute not being serialized.

Reply:
These are valid questions. 
> * Is sending the plan to the data nodes the only way of triggering serialization and deserialization via transport in ES|QL?

The serialization and deserialization are triggered by tests as well, and it is also 
triggered by cross cluster search. 

> * Are there other instances in ES|QL where we're avoiding serialization of attributes based on the knowledge that data nodes are not using that information?

If the reading and working code coordinate correctly it will work, however it can get surprise. I'd suggest to avoid serializing unnecessary things when possible, as we came across issues(OOM) with serializing large plans before, `isOperator` is a tiny thing, however many tiny things add together could be big, and considering we are likely to add an optional parameter to the `match` function to support match options, having a `TransportVersion` for `isOperator` seems a bit heavy. 
I wonder if it is a good idea to remove the isOperator from constructor, and differentiate `match` function from `:` operator from its source, so that `Verifier` can generate the correct message?

Reply:
> I wonder if it is a good idea to remove the isOperator from constructor, and differentiate match function from : operator from its source, so that Verifier can generate the correct message?

I see, this is something I hadn't thought of. It makes sense as this can be derived from `source`.

I've given it a shot in a0bc3c62cc38e608d056954d4d4d16ac6e5c2475, WDYT?